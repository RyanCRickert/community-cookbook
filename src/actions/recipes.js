import uuid from "uuid";
import database from "../firebase/firebase";

export const addRecipe = (recipe) => ({
  type: "ADD_RECIPE",
  recipe
});

export const startAddRecipe = (recipeData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const {
      name = "",
      ingredients = [],
      instructions = "",
      cookTime = 0,
      createdAt = 0
  } = recipeData;
    const recipe = { name, instructions, createdAt, ingredients, cookTime };

    return database.ref(`recipes`).push(recipe).then((ref) => {
      dispatch(addRecipe({
        id: ref.key,
        ...recipe
      }));
    });
  };
};

export const addIngredient = (id, ingredient) => ({
  type: "ADD_INGREDIENT",
  id,
  ingredient
});

export const startAddIngredient = (id, ingredient) => {
  return (dispatch) => {
    const {
      name = ""
    } = ingredientData;
    const ingredient = {name};

    return database.ref(`recipes/${id}/ingredients`).push(ingredient).then((ref) => {
      dispatch(addIngredient({
        iid: ref.key,
        ingredient
      }));
    });
  };
};

export const editIngredient = (id, iid, update) => {
  type:"EDIT_INGREDIENT",
  id,
  iid,
  update
};

export const startEditIngredient = (id, iid, update) => {
  return (dispatch) => {
    return database.ref(`recipes/${id}/ingredients${iid}`)
  };
};

export const removeRecipe = ({ id } = {}) => ({
  type: "REMOVE_RECIPE",
  id
});

export const startRemoveRecipe = ({ id }) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;

    return database.ref(`recipes/${id}`).remove().then(() => {
      dispatch(removeRecipe({ id }));
    });
  };
};

export const editRecipe = (id, updates) => ({
  type: "EDIT_RECIPE",
  id,
  updates
});

export const startEditRecipe = (id, updates) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;

    return database.ref(`recipes/${id}`).update(updates).then(() => {
      dispatch(editRecipe(id, updates));
    });
  };
};

export const setRecipes = (recipes) => ({
  type: "SET_RECIPES",
  recipes
});

export const startSetRecipes = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid

    return database.ref(`recipes`).once("value", (snapshot) => {
      const recipes = [];

      snapshot.forEach((childSnapshot) => {
        recipes.push({
          id: childSnapshot.key,
          ...childSnapshot.val()
        });
      });
      dispatch(setRecipes(recipes));
    });
  };
};

export const setIngredients = (id, ingredients) => ({
  type: "SET_INGREDIENTS",
  id,
  ingredients
});

export const startSetIngredients = () => {
  return (dispatch) => {
    return database.ref(`recipes/${id}/ingredients`).once("value", (snapshot) => {
      const ingredients = [];

      snapshot.forEach((childSnapshot) => {
        recipes.push({
          id: childSnapshot.key,
          ...childSnapshot.val()
        });
      });
      dispatch(setIngredients(ingredients));
    });
  };
};