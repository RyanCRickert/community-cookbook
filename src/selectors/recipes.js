import moment from "moment";

export default (recipes, {name, sortBy, startDate, endDate}) => {
    return recipes.filter((recipe) => {
        const createdAtMoment = moment(recipe.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, "day") : true ;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, "day") : true ;
        const nameMatch = recipe.name.toLowerCase().includes(name.toLowerCase());

        return startDateMatch && endDateMatch && nameMatch;
    }).sort((a , b) => {
        if (sortBy === "date") {
            return a.createdAt < b.createdAt ? 1 : -1;
        } else if (sortBy === "name") {
            return a.name > b.name ? 1 : -1;
        } else if (sortBy === "cookTime") {
            return a.cookTime < b.cookTime ? 1 : -1;
        } else if (sortBy === "feeds") {
            return a.feeds < b.feeds ? 1 : -1;
        }
    });
};