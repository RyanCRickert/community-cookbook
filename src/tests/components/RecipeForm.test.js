import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import RecipeForm from "../../components/RecipeForm";
import recipes from "../fixtures/recipes";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<RecipeForm />);
});

test("should render recipe form correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should render recipe form with recipe data", () =>{
  wrapper = shallow(<RecipeForm recipe={recipes[0]}/>);
  
  expect(wrapper).toMatchSnapshot();
});

test("should render error for invalid recipe submission", () =>{
  expect(wrapper).toMatchSnapshot();
  wrapper.find("form").at(1).simulate("submit", {
    preventDefault: () => { }
  });
  expect(wrapper.state("error").length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});

test("should set name on input change", () => {
  const value =  "New name";

  expect(wrapper).toMatchSnapshot();
  wrapper.find("input").at(0).simulate("change", {
    target: { value }
  });
  expect(wrapper.state("name")).toBe(value);
  expect(wrapper).toMatchSnapshot();
});

test("should set instruction on textarea change", () => {
  const value =  "New instruction";
  
  expect(wrapper).toMatchSnapshot();
  wrapper.find("textarea").at(0).simulate("change", {
    target: { value }
  });
  expect(wrapper.state("instructions")).toBe(value);
  expect(wrapper).toMatchSnapshot();
});

test("should set cook time if valid input", () => {
  const value =  2;

  expect(wrapper).toMatchSnapshot();
  wrapper.find("input").at(1).simulate("change", {
    target: { value }
  });
  expect(wrapper.state("cookTime")).toBe(value);
  expect(wrapper).toMatchSnapshot();
});

test("should not set cook time on invalid input", () => {
  const value =  20;

  expect(wrapper).toMatchSnapshot();
  wrapper.find("input").at(1).simulate("change", {
    target: { value }
  });
  expect(wrapper.state("cookTime")).toBe(20);
  expect(wrapper).toMatchSnapshot();
});

test("should call onSubmit prop for valid form submission", () => {
  const onSubmitSpy = jest.fn();
  wrapper = shallow(<RecipeForm recipe={recipes[0]} onSubmit={onSubmitSpy}/>);

  expect(wrapper).toMatchSnapshot();
  wrapper.find("form").at(1).simulate("submit", {
    preventDefault: () => { }
  });
  expect(wrapper.state("error")).toBe("");
  expect(onSubmitSpy).toHaveBeenLastCalledWith({
    name: recipes[0].name,
    instructions: recipes[0].instructions,
    ingredients: recipes[0].ingredients,
    createdAt: recipes[0].createdAt,
    feeds: recipes[0].feeds,
    category: recipes[0].category,
    cookTime: recipes[0].cookTime
  });
  expect(wrapper).toMatchSnapshot();
});