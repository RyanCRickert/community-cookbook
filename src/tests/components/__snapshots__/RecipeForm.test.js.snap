// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call onSubmit prop for valid form submission 1`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value="Baked Chicken"
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={45}
      />
       
      minutes
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={4}
      />
       
      people
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value="Chicken"
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <IngredientListItem
    handleEditIngredient={[Function]}
    handleRemoveIngredient={[Function]}
    key="chicken"
    name="chicken"
  />
  <IngredientListItem
    handleEditIngredient={[Function]}
    handleRemoveIngredient={[Function]}
    key="spices"
    name="spices"
  />
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value="Place into over at 350F"
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should call onSubmit prop for valid form submission 2`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value="Baked Chicken"
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={45}
      />
       
      minutes
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={4}
      />
       
      people
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value="Chicken"
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <IngredientListItem
    handleEditIngredient={[Function]}
    handleRemoveIngredient={[Function]}
    key="chicken"
    name="chicken"
  />
  <IngredientListItem
    handleEditIngredient={[Function]}
    handleRemoveIngredient={[Function]}
    key="spices"
    name="spices"
  />
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value="Place into over at 350F"
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should not set cook time on invalid input 1`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value=""
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      minute
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value=""
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should not set cook time on invalid input 2`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value=""
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={20}
      />
       
      minutes
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value=""
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should render error for invalid recipe submission 1`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value=""
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      minute
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value=""
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should render error for invalid recipe submission 2`] = `
<div
  className="form-edit"
>
  <p
    className="form__error"
  >
    Please enter a name.
  </p>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value=""
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      minute
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value=""
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should render recipe form correctly 1`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value=""
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      minute
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value=""
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should render recipe form with recipe data 1`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value="Baked Chicken"
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={45}
      />
       
      minutes
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={4}
      />
       
      people
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value="Chicken"
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <IngredientListItem
    handleEditIngredient={[Function]}
    handleRemoveIngredient={[Function]}
    key="chicken"
    name="chicken"
  />
  <IngredientListItem
    handleEditIngredient={[Function]}
    handleRemoveIngredient={[Function]}
    key="spices"
    name="spices"
  />
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value="Place into over at 350F"
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should set cook time if valid input 1`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value=""
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      minute
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value=""
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should set cook time if valid input 2`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value=""
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={2}
      />
       
      minutes
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value=""
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should set instruction on textarea change 1`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value=""
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      minute
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value=""
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should set instruction on textarea change 2`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value=""
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      minute
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value="New instruction"
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should set name on input change 1`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value=""
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      minute
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value=""
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;

exports[`should set name on input change 2`] = `
<div
  className="form-edit"
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Name of Dish"
    type="text"
    value="New name"
  />
  <div
    className="number-input"
  >
    <div>
      Cook Time :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      minute
    </div>
    <div>
      Feeds :
      <input
        className="number-input__item"
        min={1}
        onChange={[Function]}
        type="number"
        value={1}
      />
       
      person
    </div>
    <div>
      Category :
      <select
        className="number-input__item-category"
        onChange={[Function]}
        value=""
      >
        <option
          value={null}
        />
        <option
          value="Beef"
        >
          Beef
        </option>
        <option
          value="Chicken"
        >
          Chicken
        </option>
        <option
          value="Meatless"
        >
          Meatless
        </option>
        <option
          value="Other"
        >
          Other
        </option>
        <option
          value="Pork"
        >
          Pork
        </option>
      </select>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-input__ingredient"
      name="ingredient"
      placeholder="Add Ingredient"
      type="text"
    />
    <button
      className="button text-input__button"
    >
      Add Ingredient
    </button>
  </form>
  <p
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    *Please add at least one ingredient*
  </p>
  <form
    onSubmit={[Function]}
  >
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Cooking instructions"
      value=""
    />
    <div>
      <button
        className="button"
      >
         Recipe
      </button>
    </div>
  </form>
  <Link
    className="button home-button"
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;
